### PR TITLE
Fix issues in wmap plugin

### LINKED ISSUE #17717: WMAP Sites Uses IP Address as Vhost
The `wmap_sites` command populates the `Vhost` column with the IP address and not the hostname.

```
msf6 exploit(multi/http/lucee_scheduled_job) > wmap_sites -a http://metasploit.com
[*] Site created.
msf6 exploit(multi/http/lucee_scheduled_job) > wmap_sites -l
[*] Available sites
===============

     Id  Host            Vhost           Port  Proto  # Pages  # Forms
     --  ----            -----           ----  -----  -------  -------
     0   13.225.214.124  13.225.214.124  80    http   0        0


msf6 exploit(multi/http/lucee_scheduled_job) > 

```

I'm pretty sure that "Vhost" should be "metasploit.com" in this case.

### LINKED ISSUE #17718: WMAP Targets -t Crashes
I get a stack trace when trying to define a target site using `wmap_targets -t`. From the help output, I'm pretty sure it's supposed to be a URL.

```
msf6 exploit(multi/http/lucee_scheduled_job) > wmap_targets -h
[*] Usage: wmap_targets [options]
	-h 		Display this help text
	-t [urls]	Define target sites (vhost1,url[space]vhost2,url) 
	-d [ids]	Define target sites (id1, id2, id3 ...)
	-c 		Clean target sites list
	-l  		List all target sites

msf6 exploit(multi/http/lucee_scheduled_job) > wmap_targets -t http://metasploit.com
[-] Error while running command wmap_targets: PG::InvalidTextRepresentation: ERROR:  invalid input syntax for type inet: "metasploit.com"
CONTEXT:  unnamed portal parameter $2 = '...'


Call stack:
/home/smcintyre/.rvm/gems/ruby-3.0.4@metasploit-framework/gems/activerecord-7.0.4.2/lib/active_record/connection_adapters/postgresql_adapter.rb:783:in `exec_prepared'
/home/smcintyre/.rvm/gems/ruby-3.0.4@metasploit-framework/gems/activerecord-7.0.4.2/lib/active_record/connection_adapters/postgresql_adapter.rb:783:in `block (2 levels) in exec_cache'
/home/smcintyre/.rvm/gems/ruby-3.0.4@metasploit-framework/gems/activesupport-7.0.4.2/lib/active_support/concurrency/share_lock.rb:187:in `yield_shares'

...
```

### PR DESCRIPTION
**Pull request by dwelch-r7**
Resolves #17718, resolves #17717

This PR resolves two issues in the wmap plugin related to the `wmap_targets -t` and the `wmap_sites -a` commands which had _similar_ issues

First up `wmap_targets -t`:
Intuitively you'd expect to use this as follows `wmap_targets -t http://my.cool.website` but that results in the stacktrace described in #17718 
The intended use is supposed to be `wmap_targets -t http://1.2.3.4` or `wmap_targets -t my.cool.website,1.2.3.4` with the option for separating multiple targets with spaces
https://github.com/rapid7/metasploit-framework/blob/71699235998a1aa4291bf3af25c7b14871436864/plugins/wmap.rb#L1338-L1340

Obviously not great, so I've kept compatibility here with the old method while allowing the more intuitive uses cases to also work

# Verification Steps
- [x] Run `msfconsole`
- [x] Load the wmap plugin `load wmap`
- [x] Ensure old usage continues to work: `wmap_targets -t http://1.2.3.4`, `wmap_targets -t my.cool.website,1.2.3.4`
- [x] Verify with `wmap_targets -l`
- [ ] Ensure the new usage works as expected: `wmap_targets -t http://my.cool.website` 
- [ ] Verify with `wmap_targets -l`

Second up is `wmap_sites -a`:
This issue described in #17717 is that when adding a site via it's address like `wmap_sites -a http://my.cool.website` when listing the added sites with `wmap_sites -l` it would appear with the vhost set incorrectly to the ip instead of `my.cool.website`
Similar to the first issue you could run it like `wmap_sites -a my.cool.website,http://1.2.3.4` and the vhost would be set to `my.cool.website` but this PR adds the correct vhost when specifying a url normally
https://github.com/rapid7/metasploit-framework/blob/71699235998a1aa4291bf3af25c7b14871436864/plugins/wmap.rb#L1275-L1279

# Verification Steps
- [ ] Run `msfconsole`
- [ ] Load the wmap plugin `load wmap`
- [ ] Ensure old usage continues to work: `wmap_sites -a http://1.2.3.4`, `wmap_sites -a my.cool.website,1.2.3.4`
- [ ] Verify with `wmap_sites -l`
- [ ] Ensure the new usage works as expected: `wmap_sites -a http://my.cool.website` 
- [ ] Verify with `wmap_sites -l`

### CODE REVIEW

**Comment by dwelch-r7** [comment]
> The code looks good, but when I try one of your examples, I'm getting another stack trace now. It hangs for a long while, probably while it's trying to resolve the name before crashing.
> 
> ```
> msf6 exploit(windows/smb/psexec) > wmap_sites -a http://my.cool.website
> [-] Error while running command wmap_sites: getaddrinfo: Temporary failure in name resolution
> 
> Call stack:
> /home/smcintyre/.rvm/gems/ruby-3.0.4@metasploit-framework/gems/rex-socket-0.1.47/lib/rex/socket.rb:193:in `getaddrinfo'
> /home/smcintyre/.rvm/gems/ruby-3.0.4@metasploit-framework/gems/rex-socket-0.1.47/lib/rex/socket.rb:193:in `getaddresses'
> /home/smcintyre/.rvm/gems/ruby-3.0.4@metasploit-framework/gems/rex-socket-0.1.47/lib/rex/socket.rb:175:in `getaddress'
> /home/smcintyre/Repositories/metasploit-framework.pr/lib/msf/core/db_manager/web.rb:215:in `block in report_web_site'
> /home/smcintyre/.rvm/gems/ruby-3.0.4@metasploit-framework/gems/activerecord-7.0.4.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:215:in `with_connection'
> /home/smcintyre/Repositories/metasploit-framework.pr/lib/msf/core/db_manager/web.rb:192:in `report_web_site'
> /home/smcintyre/Repositories/metasploit-framework.pr/lib/metasploit/framework/data_service/proxy/web_data_proxy.rb:5:in `block in report_web_site'
> /home/smcintyre/Repositories/metasploit-framework.pr/lib/metasploit/framework/data_service/proxy/core.rb:164:in `data_service_operation'
> /home/smcintyre/Repositories/metasploit-framework.pr/lib/metasploit/framework/data_service/proxy/web_data_proxy.rb:4:in `report_web_site'
> /home/smcintyre/Repositories/metasploit-framework.pr/plugins/wmap.rb:1321:in `add_web_site'
> /home/smcintyre/Repositories/metasploit-framework.pr/plugins/wmap.rb:135:in `cmd_wmap_sites'
> /home/smcintyre/Repositories/metasploit-framework.pr/lib/rex/ui/text/dispatcher_shell.rb:581:in `run_command'
> /home/smcintyre/Repositories/metasploit-framework.pr/lib/rex/ui/text/dispatcher_shell.rb:530:in `block in run_single'
> /home/smcintyre/Repositories/metasploit-framework.pr/lib/rex/ui/text/dispatcher_shell.rb:524:in `each'
> /home/smcintyre/Repositories/metasploit-framework.pr/lib/rex/ui/text/dispatcher_shell.rb:524:in `run_single'
> /home/smcintyre/Repositories/metasploit-framework.pr/lib/rex/ui/text/shell.rb:168:in `run'
> /home/smcintyre/Repositories/metasploit-framework.pr/lib/metasploit/framework/command/console.rb:48:in `start'
> /home/smcintyre/Repositories/metasploit-framework.pr/lib/metasploit/framework/command/base.rb:82:in `start'
> ./msfconsole:23:in `<main>'
> ```

oh yea that was just like a placeholder/example and not meant to actually be tried....but happy accident since it probably still shouldn't crash though, I wouldn't have expected that, I thought the only exception raised by `Rex::Socket.getaddress` would be a SocketError but I guess I'm mistaken, I'll dig a bit and update it to handlethis new exception

File: plugins/wmap.rb [hunk 1]
```diff
@@ -1316,7 +1318,13 @@ def add_web_site(url)
           ssl = true
         end
 
-        site = framework.db.report_web_site(wait: true, host: uri.host, port: uri.port, vhost: vhost, ssl: ssl, workspace: framework.db.workspace)
+        site = begin
+          framework.db.report_web_site(wait: true, host: uri.host, port: uri.port, vhost: vhost, ssl: ssl, workspace: framework.db.workspace)
+        rescue SocketError => e
```
Comment (sempervictus) [inline plugins/wmap.rb:1323]: Upstream's logic behind making a layer7 protocol failure (DNS resolution) present as a layer 4 condition aside, this ought to still work correctly with the native resolver (https://github.com/rapid7/rex-socket/pull/43) code as the exception that throws emulates this one from the standard library.

**Comment by smcintyre-r7** [comment]
Everything is looking good to me now. I didn't find any issues while going through and testing it.

<details>
<summary>Testing Output</summary>

```
msf6 payload(cmd/unix/python/pingback_bind_tcp) > wmap_sites -a http://my.cool.website
[*] Could not get address for my.cool.website.
[-] Unable to create site
msf6 payload(cmd/unix/python/pingback_bind_tcp) > wmap_sites -a my.cool.website,1.2.3.4
[*] Site created.
msf6 payload(cmd/unix/python/pingback_bind_tcp) > wmap_sites -l
[*] Available sites
===============

     Id  Host     Vhost            Port  Proto  # Pages  # Forms
     --  ----     -----            ----  -----  -------  -------
     0   1.2.3.4  my.cool.website  80    http   0        0


msf6 payload(cmd/unix/python/pingback_bind_tcp) > workspace -D
[*] Deleted workspace: default
[*] Recreated the default workspace
msf6 payload(cmd/unix/python/pingback_bind_tcp) > wmap_sites -l
[*] Available sites
===============

     Id  Host  Vhost  Port  Proto  # Pages  # Forms
     --  ----  -----  ----  -----  -------  -------


msf6 payload(cmd/unix/python/pingback_bind_tcp) > wmap_sites -a http://1.2.3.4
[*] Site created.
msf6 payload(cmd/unix/python/pingback_bind_tcp) > wmap_sites -l
[*] Available sites
===============

     Id  Host     Vhost    Port  Proto  # Pages  # Forms
     --  ----     -----    ----  -----  -------  -------
     0   1.2.3.4  1.2.3.4  80    http   0        0


msf6 payload(cmd/unix/python/pingback_bind_tcp) > wmap_sites -a http://metasploit.com
[*] Site created.
msf6 payload(cmd/unix/python/pingback_bind_tcp) > wmap_sites -l
[*] Available sites
===============

     Id  Host         Vhost           Port  Proto  # Pages  # Forms
     --  ----         -----           ----  -----  -------  -------
     0   1.2.3.4      1.2.3.4         80    http   0        0
     1   18.67.65.65  metasploit.com  80    http   0        0


msf6 payload(cmd/unix/python/pingback_bind_tcp) > 
```
</details>

**Review by smcintyre-r7 — APPROVED** [review]

**Comment by adfoster-r7** [comment]
## Release Notes

Updates the wmap plugin to no longer crash when running `wmap_targets -t http://metasploit.com`

**Pull request merged**